NewForce-Cohort-6/tabloidfullstack-cranberry-sauce
# Ticket #11 - Delete a Category

# Description
This PR is for [Ticket #11](https://trello.com/c/2CYJNpJb/11-11-delete-a-category), which should allow users to select and delete a category from the list of categories.

> As an admin I would like to be able to delete a Category so that I can remove any that are not needed.
> 
> **Given** an admin is vising the list of Categories
> **When** they select the `Delete` option
> **Then** they should be prompted to confirm the deletion
> 
> **Given** the admin wishes to confirm the delete
> **When** they the select the option to confirm
> **Then** the Category should be removed from the system
> **And** the admin should be directed back to the Category list
> 
> **Given** the admin decides not to confirm the delete
> **When** they the select the option to reject confirmation
> **Then** the Category should NOT be removed from the system
> **And** the admin should be directed back to the Category list
> 
> > **NOTE:** For the time being it is acceptable to treat all users as `admin` users. There is a future story about enforcing user permissions.

# Testing Instructions
1. Commit or stash whatever you're currently working on
2. Git fetch --all
3. Git checkout ac-deleteCategory
4. If you don't see the most recent version of my files, git pull origin ac-deleteCategory
5.  If you need to create the Tabloid database, [use this file to do so](https://github.com/NewForce-Cohort-6/tabloidfullstack-cranberry-sauce/blob/main/SQL/01_Tabloid_Create_DB.sql)
6. If you don't have the Tabloid seed data, [use this file to get it
](https://github.com/NewForce-Cohort-6/tabloidfullstack-cranberry-sauce/blob/main/SQL/02_Tabloid_Seed_Data.sql)
7. To make sure that there is at least one category without a post attached, add one or all of these to your database:
`INSERT INTO Category([Name])
VALUES('Paranormal')`
`INSERT INTO Category([Name])
VALUES ('Humor')`
`INSERT INTO Category([Name])
VALUES ('Outdoors)`
8. Execute Tabloid from inside of Visual Studio
![Tabloid](https://user-images.githubusercontent.com/106983427/202474349-d57baf9b-323c-4406-b75a-d8f99a886603.png)
9.  Run NPM Start from inside of the Tabloid client folder
![client](https://user-images.githubusercontent.com/106983427/202473962-42d0b50f-1420-4b34-a522-464dc9fc11d5.png)
10. Login using an email that you know is there and works for you (example: foo@bar.com from seed data)
11. Once logged-in, use the navbar to navigate to category management ("/category")
12. Click the delete button located next to the name of each category that doesn't have an associated post
13. You should be redirected to a page ("/category/delete/id") that asks for confirmation
14. Click delete to confirm that you want to remove the selected category 
15. View the list of categories to confirm that the category was removed

> **NOTE:** For the time being, I pulled Post.cs from Luke's pull request so that I could use the information for conditional rendering. Because it's identical, there shouldn't be any conflicts. If conflicts arise, I will return to fix them.

# Checklist:
